drivers: sensor: add wurth elektronik sensor drivers

wm-eisos opened the pull request:
This pull request updates the revision of the hal, removes old drivers and updates and adds new drivers.

@mah-eiSmart

wm-eisos committed:
drivers: sensor: wsen_itds: remove wsen_itds driver

Remove wsen_itds driver since the hal update
is no longer compatible with this version.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_hids: remove wsen_hids driver

Remove wsen_hids since the hal update
is no longer compatible with this version.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_pads: remove wsen_pads driver

Remove wsen_pads since the hal update
is no longer compatible with this version.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_pdus: remove wsen_pdus driver

Remove wsen_pdus since the hal update
is no longer compatible with this version.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_tids: remove wsen_tids driver

Remove wsen_tids since the hal update
is no longer compatible with this version.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_hids_2525020210002: add sensor driver

Add wsen_hids_2525020210002 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_pdus_25131308XXXXX: add sensor driver

Add wsen_pdus_25131308XXXXX driver with
the corrected name and compatibility with
the hal update as well as added new features..

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_tids_2521020222501: add sensor driver

Add wsen_tids_2521020222501 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_hids_2525020210001: add sensor driver

Add wsen_hids_2525020210001 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_pads_2511020213301: add sensor driver

Add wsen_pads_2511020213301 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_itds_2533020201601: add sensor driver

Add wsen_itds_2533020201601 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
drivers: sensor: wsen_isds_2536030320001: add sensor driver

Add wsen_isds_2536030320001 driver with
the corrected name and compatibility with
the hal update as well as added new features.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos committed:
manifest: hal_wurthelektronik: update to latest

Update hal_wurthelektronik to include latest sensor core drivers.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

mah-eiSmart commented:
This is a big improvement to the Würth Elektronik sensor drivers, as

- the sensor names are correct now (the name without article number does not identify the sensor)
- numerous useful functions have been added
- examples are added how to use the custom attributes
- it's using the newest würth elektronik hal with the latest bug fixes

Thank you for contributing @wm-eisos and for reviewing.

BTW: The examples and drivers have been tested on several hardware platforms.

wm-eisos commented:
@yperess can you please rereview it.

Thanks in advance

yperess commented:
> @yperess can you please rereview it.
> 
> Thanks in advance

Looks like a few extra compliance tests failing still

MaureenHelm commented on drivers/sensor/CMakeLists.txt:
```diff
@@ -58,7 +58,6 @@ add_subdirectory_ifdef(CONFIG_INA3221 ina3221)
 add_subdirectory_ifdef(CONFIG_IST8310 ist8310)
 add_subdirectory_ifdef(CONFIG_ISL29035 isl29035)
 add_subdirectory_ifdef(CONFIG_ISM330DHCX ism330dhcx)
-add_subdirectory_ifdef(CONFIG_ITDS wsen_itds)
```
Please explain in the commit message why you're removing the driver

MaureenHelm replied:
Would like to see more explanation than just "we're removing the driver to replace it with a new one". You mention the reason in https://github.com/zephyrproject-rtos/zephyr/pull/62408#issuecomment-1711566468, but would like to have that captured in the commit history.

wm-eisos replied:
i changed the commit message to indicate that the new hal is not compatible with the old drivers.

MaureenHelm commented on drivers/sensor/wsen_hids_2525020210002/Kconfig:
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2023 Würth Elektronik eiSos GmbH & Co. KG
+# SPDX-License-Identifier: Apache-2.0
+
+config WSEN_HIDS_2525020210002
```
Is `2525020210002` a part number or SKU? This is a long name for a driver

wm-eisos replied:
This is a part number uniquely identifying the sensor as for example hids refers to the family of the sensor (humidity in this case), so it is important to have it part of the driver name.

MaureenHelm commented on drivers/sensor/wsen_hids_2525020210002/Kconfig:
```diff
@@ -0,0 +1,11 @@
+	bool "WSEN-HIDS-2525020210002 humidity sensor"
+	default y
+	depends on DT_HAS_WE_WSEN_HIDS_2525020210002_ENABLED
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_WE_WSEN_HIDS_2525020210002),i2c)
```
Does this device support SPI or any other non-I2C interface? If not, you can remove the conditional if

MaureenHelm commented on drivers/sensor/wsen_hids_2525020210002/wsen_hids_2525020210002.c:
```diff
@@ -0,0 +1,255 @@
+		.precision = (hids_2525020210002_precision_t)(DT_INST_ENUM_IDX(inst, precision)),  \
+		.heater = (hids_2525020210002_heater_t)(DT_INST_ENUM_IDX(inst, heater)),           \
+	};                                                                                         \
+	HIDS_2525020210002_DEVICE_INIT(inst)
```
```suggestion
	SENSOR_DEVICE_DT_INST_DEFINE(inst, hids_2525020210002_init, NULL,                          \
				     &hids_2525020210002_data_##inst,                              \
				     &hids_2525020210002_config_##inst, POST_KERNEL,               \
				     CONFIG_SENSOR_INIT_PRIORITY, &hids_2525020210002_driver_api);
```

wm-eisos replied:
this has been applied to all of the drivers.

MaureenHelm commented on dts/bindings/sensor/we,wsen-hids-2525020210002.yaml:
```diff
@@ -0,0 +1,37 @@
+
+  precision:
+    type: string
+    default: "Medium"
```
The default value needs to be justified in the description. For example, is it a hardware register reset value?

wm-eisos replied:
default value explanation has been added.

MaureenHelm commented on dts/bindings/sensor/we,wsen-hids-2525020210002.yaml:
```diff
@@ -0,0 +1,37 @@
+
+  heater:
+    type: string
+    default: "Off"
```
Same here

wm-eisos replied:
default value explanation has been added.

MaureenHelm commented on samples/sensor/wsen_hids_2525020210002/README.rst:
```diff
@@ -0,0 +1,40 @@
+Requirements
+************
+
+ This sample uses the WSEN-HIDS sensor controlled using the I2C interface.
```
We're trying to stop introducing new driver-specific samples and move toward having more generic samples for each sensor type. Please look at `samples/sensor/accel_polling` for example

wm-eisos replied:
The samples for the drivers has been supplied because some of the drivers implement custom attribues or triggers. which without looking at the example would be really hard for the user to use.

MaureenHelm replied:
Can we postpone the new samples to another PR?

wm-eisos replied:
i just removed them from this PR and will create a seperate one affter this one is merged.

MaureenHelm reviewed (changes requested):
This would be easier to review and get merged if you break it up into multiple PRs. Each PR should just cover one driver, removing the old one and adding the new one.

wm-eisos commented:
> This would be easier to review and get merged if you break it up into multiple PRs. Each PR should just cover one driver, removing the old one and adding the new one.

They have all been done in one PR since in this PR the halhas been updated with breaking changes to the old drivers and the new drivers need the new hal to function corectly.

wm-eisos commented:
@yperess i see that a couple of checks have been stuck for a while now. Do these require a review before they trigger?

wm-eisos commented:
> > @yperess i see that a couple of checks have been stuck for a while now. Do these require a review before they trigger?
> 
> Approved the workflow

Can you please reapprove it again now, i rann the check compliance py script this time and solved the issues hopefully no more compliance check fails.

wm-eisos commented:
@yperess @MaureenHelm since all checks have passed this time. can you please review it before any merge conflicts could happen.
Thanks in advance

wm-eisos commented:
@yperess @MaureenHelm this pull request is now ready for review

wm-eisos commented:
@MaureenHelm @yperess i just resolved the merge conflicts again. please review the commits before another merge conflict occurs.

wm-eisos commented:
@yperess can you please approve the workflow

wm-eisos commented:
Should i open another PR with the same commits as it seems to appear that this pull request has been inactive for 2 weeks

@MaureenHelm @yperess @nashif @teburd  @tristan-google

kartben commented:
> Should i open another PR with the same commits as it seems to appear that this pull request has been inactive for 2 weeks
> 
> @MaureenHelm @yperess @nashif @teburd  @tristan-google 

I've triggered the workflow, there should be no need to re-open a PR since you already did so yourself. FYI given we are now in feature freeze for 3.5 this won't be another week until it can be considered for merging though.

wm-eisos commented:
@kartben can you please retrigger it

MaureenHelm commented on drivers/sensor/wsen_hids_2525020210002/wsen_hids_2525020210002.c:
```diff
@@ -0,0 +1,248 @@
+	/* Initialize WE sensor interface */
+	HIDS_Get_Default_Interface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = WE_i2c;
+	data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
```
Need to check if the I2C bus is ready with  `i2c_is_ready_dt` before using it.

wm-eisos replied:
For every driver now the correct is ready function is called depending on the interface

MaureenHelm commented on drivers/sensor/CMakeLists.txt:
```diff
@@ -147,6 +147,7 @@ add_subdirectory_ifdef(CONFIG_VEML7700 veml7700)
 add_subdirectory_ifdef(CONFIG_VL53L1X vl53l1x)
 add_subdirectory_ifdef(CONFIG_VOLTAGE_DIVIDER voltage_divider)
+add_subdirectory_ifdef(CONFIG_WSEN_PDUS_25131308XXXXX		wsen_pdus_25131308XXXXX)
 add_subdirectory_ifdef(CONFIG_WSEN_HIDS_2525020210002		wsen_hids_2525020210002)
```
Please keep these in alpha order

wm-eisos replied:
Fixed

MaureenHelm commented on drivers/sensor/Kconfig:
```diff
@@ -203,6 +203,7 @@ source "drivers/sensor/veml7700/Kconfig"
 source "drivers/sensor/vl53l1x/Kconfig"
 source "drivers/sensor/voltage_divider/Kconfig"
+source "drivers/sensor/wsen_pdus_25131308XXXXX/Kconfig"
 source "drivers/sensor/wsen_hids_2525020210002/Kconfig"
```
Please keep these in alpha order

MaureenHelm commented on drivers/sensor/wsen_pdus_25131308XXXXX/wsen_pdus_25131308XXXXX.c:
```diff
@@ -0,0 +1,160 @@
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP: {
+		int32_t temperatureMega = ((int32_t)(data->temperature - T_MIN_VAL_PDUS)) * 4272;
```
Are the magic numbers here and throughout this function documented somewhere?

wm-eisos replied:
I have added a comment that says that these value are mentioned in the user manual

MaureenHelm commented on drivers/sensor/wsen_pdus_25131308XXXXX/wsen_pdus_25131308XXXXX.c:
```diff
@@ -0,0 +1,160 @@
+	switch (data->sensor_interface.interfaceType) {
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+	case WE_i2c:
+		data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
```
Need to call `i2c_is_ready_dt`

wm-eisos replied:
Fixed

MaureenHelm commented on dts/bindings/sensor/we,wsen-pdus-25131308XXXXX.yaml:
```diff
@@ -0,0 +1,22 @@
+    - 2 # order code 2513130810201, range = -10 to +10 kPa
+    - 3 # order code 2513130810301, range =  0 to 100 kPa
+    - 4 # order code 2513130810401 (5V VCC) + 2513130810402 (3.3V VCC), range = -100 to +1000 kPa
+    - 5 # order code 2513130815401, range = 0 to +1500 kPa
```
Move the details into the description field so it will get rendered in the [documentation](https://docs.zephyrproject.org/latest/build/dts/api/bindings.html)

MaureenHelm commented on drivers/sensor/wsen_tids_2521020222501/wsen_tids_2521020222501.c:
```diff
@@ -0,0 +1,330 @@
+
+	struct tids_2521020222501_data *data = dev->data;
+
+	TIDS_state_t continuousModeState;
```
Please no camel case

wm-eisos replied:
I made all variables of the form lower case with underscores between different nouns

MaureenHelm commented on drivers/sensor/wsen_tids_2521020222501/wsen_tids_2521020222501_trigger.c:
```diff
@@ -0,0 +1,304 @@
+static void tids_2521020222501_process_interrupt(const struct device *dev)
+{
+	struct tids_2521020222501_data *data = dev->data;
+	struct sensor_trigger temperature_low_trigger = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_TIDS_2521020222501_THRESHOLD_LOWER};
+	struct sensor_trigger temperature_high_trigger = {
+		.chan = SENSOR_CHAN_AMBIENT_TEMP,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_TIDS_2521020222501_THRESHOLD_UPPER};
```
Need to save a pointer to the user's sensor trigger and pass it back, rather than create a new sensor trigger on the stack. See #55130

wm-eisos replied:
The trigger pointer is now stored

MaureenHelm commented on drivers/sensor/wsen_pads_2511020213301/wsen_pads_2511020213301_trigger.c:
```diff
@@ -0,0 +1,345 @@
+	struct pads_2511020213301_data *data = dev->data;
+
+#ifdef CONFIG_WSEN_PADS_2511020213301_PRESSURE_THRESHOLD
+	struct sensor_trigger pressure_low_trigger = {
+		.chan = SENSOR_CHAN_PRESS,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_PADS_2511020213301_THRESHOLD_LOWER};
+	struct sensor_trigger pressure_high_trigger = {
+		.chan = SENSOR_CHAN_PRESS,
+		.type = (enum sensor_trigger_type)
+			SENSOR_TRIG_WSEN_PADS_2511020213301_THRESHOLD_UPPER};
```
Don't create new trigger structs on the stack; store a pointer to the user's sensor trigger and pass it back

MaureenHelm commented on drivers/sensor/wsen_itds_2533020201601/wsen_itds_2533020201601_trigger.c:
```diff
@@ -0,0 +1,562 @@
+	}
+
+#ifdef CONFIG_WSEN_ITDS_2533020201601_TAP
+	struct sensor_trigger single_tap_trigger = {.chan = SENSOR_CHAN_ALL,
+						    .type = SENSOR_TRIG_TAP};
```
Don't create new trigger structs on the stack; store a pointer to the user's sensor trigger and pass it back

MaureenHelm commented on tests/drivers/build_all/sensor/sensors_trigger_global.conf:
```diff
@@ -53,5 +53,8 @@ CONFIG_TMD2620_TRIGGER_GLOBAL_THREAD=y
 CONFIG_VCNL4040_TRIGGER_GLOBAL_THREAD=y
-CONFIG_WSEN_HIDS_TRIGGER_GLOBAL_THREAD=y
-CONFIG_WSEN_TIDS_TRIGGER_GLOBAL_THREAD=y
+CONFIG_WSEN_TIDS_2521020222501_TRIGGER_GLOBAL_THREAD=y
+CONFIG_WSEN_HIDS_2525020210001_TRIGGER_GLOBAL_THREAD=y
+CONFIG_WSEN_PADS_2511020213301_TRIGGER_GLOBAL_THREAD=y
+CONFIG_WSEN_ITDS_2533020201601_TRIGGER_GLOBAL_THREAD=y
+CONFIG_WSEN_ISDS_2536030320001_TRIGGER_GLOBAL_THREAD=y
```
Please keep alpha order

wm-eisos commented:
please retrigger the workflow i just marked the PR as ready for review

wm-eisos commented:
@MaureenHelm any updates regarding the rereview?

wm-eisos commented:
Latest force push was to apply 191865b on the drivers that have triggers

wm-eisos commented:
@kartben @MaureenHelm can you please review the changes?

mah-eiSmart commented:
Hello @carlescufi , hello @MaureenHelm ,

please approve this pull request. People are waiting to get access to the new version of the drivers.

Thank you for the support.

MaureenHelm commented:
@teburd @yperess can I get your help reviewing this please?

wm-eisos commented:
@MaureenHelm any updates regarding this?

wm-eisos commented:
@kartben @yperess can you guys please check out this PR?

wm-eisos commented:
@MaureenHelm can you please check it out. im not sure if i should rebase,resolve the conflict and force push again.

wm-eisos commented:
@kartben any updates please?

kartben commented on samples/sensor/wsen_hids/README.rst:
All the removed samples need entries added to `doc\_scripts\redirects.py` so that e.g. https://docs.zephyrproject.org/3.5.0/samples/sensor/wsen_hids/README.html takes people to a page that exists when they click on "The latest development version of this page may be more current than this released x.y.z version."
You'll want to redirect to either samples/sensor or whichever "generic" sample makes the most sense for each.

kartben commented:
> @MaureenHelm @kartben can you please check it out. im not sure if i should rebase, resolve the conflict and force push again.

This will dismiss existing reviews but yes, this will need to be done eventually, i.e new sensors starting at 0x81 at the time of writing

wm-eisos commented:
> > @MaureenHelm @kartben can you please check it out. im not sure if i should rebase, resolve the conflict and force push again.
> 
> This will dismiss existing reviews but yes, this will need to be done eventually, i.e new sensors starting at 0x81 at the time of writing

any solution regarding the times to get the PRs approved it has been a cycle of me rebasing and fixing conflicts and before the PR is merged there is already another conflict

kartben commented:
> > > @MaureenHelm @kartben can you please check it out. im not sure if i should rebase, resolve the conflict and force push again.
> > 
> > 
> > This will dismiss existing reviews but yes, this will need to be done eventually, i.e new sensors starting at 0x81 at the time of writing
> 
> any solution regarding the times to get the PRs approved it has been a cycle of me rebasing and fixing conflicts and before the PR is merged there is already another conflict

Not really a criticism of your approach here, maybe there was no other way than to lump everything under the same PR (?), but frankly I think this boils down to the change being thousands of lines, which is incredibly much harder to manage for everyone involved, including reviewers.

See https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow -- "One general practice we encourage, is to make small, controlled changes. This practice simplifies review, makes merging and rebasing easier, and keeps the change history clear and clean."

For example, I am about to provide you with a few comments of things that need fixing, but as you'll have addressed them, that still won't be enough for me to give a +1 as the rest of the PR is beyond my domain of expertise.

wm-eisos commented:
> > > > @MaureenHelm @kartben can you please check it out. im not sure if i should rebase, resolve the conflict and force push again.
> > > 
> > > 
> > > This will dismiss existing reviews but yes, this will need to be done eventually, i.e new sensors starting at 0x81 at the time of writing
> > 
> > 
> > any solution regarding the times to get the PRs approved it has been a cycle of me rebasing and fixing conflicts and before the PR is merged there is already another conflict
> 
> Not really a criticism of your approach here, maybe there was no other way than to lump everything under the same PR (?), but frankly I think this boils down to the change being thousands of lines, which is incredibly much harder to manage for everyone involved, including reviewers.
> 
> See https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow -- "One general practice we encourage, is to make small, controlled changes. This practice simplifies review, makes merging and rebasing easier, and keeps the change history clear and clean."
> 
> For example, I am about to provide you with a few comments of things that need fixing, but as you'll have addressed them, that still won't be enough for me to give a +1 as the rest of the PR is beyond my domain of expertise.

The thing is in this PR there is a hal update which produces breaking changes for the older drivers. Thats why i have mutliple drivers in the same PR.

kartben commented on dts/bindings/sensor/we,wsen-itds.yaml:
If you're dropping / renaming drivers, this needs to be documented in the release notes and migration guide as this will be requiring users update their code, and they need to know what to do

wm-eisos replied:
Is there a guide on which files need to be modified?

kartben replied:
doc/releases/release-notes-3.6.rst
doc/releases/migration-guide-3.6.rst

kartben commented:
> The thing is in this PR there is a hal update which produces breaking changes for the older drivers. 

So are you saying that just taking https://github.com/zephyrproject-rtos/zephyr/pull/62408/commits/fd5cecd7b4168079c3c2ecdc5784fc728832d43f leads to a configuration where drivers don't compile? We try to keep things bissectable so it would be good to fix if needed.

> Thats why i have mutliple drivers in the same PR.

Yup but for example dropping the "non-generic" samples could be an entirely different PR, completely independent of the HAL bump, right? And this one would likely take a lot less longer to review.

kartben commented:
> Regarding the first point that is correct if the commit is merged then yes the older drivers wont compile.

Then please squash accordingly so that all commits correspond to a working state.

wm-eisos commented:
> > Regarding the first point that is correct if the commit is merged then yes the older drivers wont compile.
> 
> Then please squash accordingly so that all commits correspond to a working state.

what exactly should i squash

wm-eisos commented:
depends on PR (#68462) to be merged

wm-eisos committed:
doc: releases: update notes for wurth drivers update

update release notes and migration guide for
wurth elektronik sensor drivers update.

Signed-off-by: Wajdi ELMuhtadi <wajdi.elmuhtadi@we-online.com>

wm-eisos commented:
@kartben regarding this PR, it is currently in draft but since the commits are no longer in this PR. This PR will fail because it still has the examples for the removed drivers? Anything to do regarding that ? i would like both to be merged for RC1.

kartben commented:
@wm-eisos fyi and just to set expectations, we have entered feature freeze for Zephyr 3.6 release so it might be simpler to hold off for now ; in particular, the migration guide for 3.7 doesn't exist just yet so it will be easier for you to add to it once it's been seeeded after 3.6 is out. Thanks!

wm-eisos commented:
any updates regarding this PR ?
@kartben @yperess @aescolar

mah-eiSmart commented:
> This pull request has been marked as stale because it has been open (more than) 60 days with no activity. Remove the stale label or add a comment saying that you would like to have the label removed otherwise this pull request will automatically be closed in 14 days. Note, that you can always re-open a closed pull request at any time.

Please remove the stale label. We are still waiting for review and merge.

aescolar commented:
> Please remove the stale label. We are still waiting for review and merge.

@mah-eiSmart You will need to rebase the PR as it is currently conflicting with main.
@MaureenHelm, @kartben could you take another look?

aescolar commented:
@wm-eisos ^^

mah-eiSmart commented:
> > Please remove the stale label. We are still waiting for review and merge.
> 
> @mah-eiSmart You will need to rebase the PR as it is currently conflicting with main. @MaureenHelm, @kartben could you take another look?

@aescolar , yes that's right. My colleague, which is back next week, will do this. Actually, as you see in the history, we rebased already many times. After rebase, it would be nice to get finally the approval before the next incompatible changes of the master make the review again void.

wm-eisos commented:
> @wm-eisos ^^

I have rebased the PR yesterday. please try to take another look as soon as possible.

aescolar commented:
@MaureenHelm, @kartben could you take another look at this PR?

decsny reviewed (dismissed):
are all of these bindings really needed at this point

wm-eisos commented:
> are all of these bindings really needed at this point

Yes as these bindings are required to setup the sensor acccording to the behaviour that the user wants. PS this PR contains drivers for multiple sensors 7 in total

wm-eisos commented:
Can you please remove the Stale label, i have rebased the PR can you please check it as well.

decsny commented:
there is a merge conflict

wm-eisos commented:
> there is a merge conflict

I will solve it now however as you can see in this PR this loop keeps hapening where the merge conflicts are resolved and then new merge conflicts come up because the PR is still not merged

decsny commented:
> > there is a merge conflict
> 
> I will solve it now however as you can see in this PR this loop keeps hapening where the merge conflicts are resolved and then new merge conflicts come up because the PR is still not merged

welcome to my life as well

decsny commented:
@mah-eiSmart it looks like you approved of this PR, is that still the case

decsny commented on dts/bindings/sensor/we,wsen-hids-2525020210001-i2c.yaml:
can you maybe put all these bindings in a subfolder, it's pretty noisy to have all of these in the top sensor binding folder

wm-eisos replied:
As i currently see no other sensor goups its bindings also should i group the bindings per sensor or per vendor. And if you have an example ffor a sensor with grouped bindings that would be great

decsny replied:
it doesn't matter if there's no examples like that, and the structure of the bindings folders doesnt functionally affect anything, you could move these bindings to the wifi binding folder and it would not make any difference except to humans looking at the folders being confused why that would be there. so, it would help to limit the noise by putting these associated bindings in a subfolder IMO

decsny replied:
and to answer your question, i think you should group it by sensor, not vendor

wm-eisos replied:
okay will do

mah-eiSmart commented:
> @mah-eiSmart it looks like you approved of this PR, is that still the case

Yes, from my side it's still fine!

decsny reviewed (approved):
somebody with sensor knowledge should review

dkalowsk commented:
> somebody with sensor knowledge should review

@MaureenHelm as sensor maintainer, are you able to take a review of this?

mah-eiSmart commented:
> > somebody with sensor knowledge should review
> 
> @MaureenHelm as sensor maintainer, are you able to take a review of this?

@dkalowsk Seems, @MaureenHelm is busy. Can we go forward without her help?

dkalowsk commented:
> > > somebody with sensor knowledge should review
> > 
> > 
> > @MaureenHelm as sensor maintainer, are you able to take a review of this?
> 
> @dkalowsk Seems, @MaureenHelm is busy. Can we go forward without her help?

There are ways to do so.  You can escalate in the developer meeting or on discord. I’m a little more concerned that this PR is a year in the making and stalling out on reviews. @nashif @MaureenHelm ping.

mah-eiSmart commented:
> > > > somebody with sensor knowledge should review
> > > 
> > > 
> > > @MaureenHelm as sensor maintainer, are you able to take a review of this?
> > 
> > 
> > @dkalowsk Seems, @MaureenHelm is busy. Can we go forward without her help?
> 
> There are ways to do so. You can escalate in the developer meeting or on discord. I’m a little more concerned that this PR is a year in the making and stalling out on reviews. @nashif @MaureenHelm ping.

@dkalowsk , this PR has been adopted to new versions of Zephyr many times, in the past 12 months. Thus it's still up to date. Yesterday I got feedback that customers did not choose that sensor products, because the drivers are not up to date. Escalating on discord or on developer meeting we did not do so far, we have no experience with it.

wm-eisos commented:
This PR will be closed as the plan is to have seperate PRs per sensor
